### PR TITLE
Fix Color normalization in Replay

### PR DESCRIPTION
**Pull request by Kalabint**
_Removed hardcoded 0 in minSpeed, its now fully dynamic._

Before:

image

After:

image

### CODE REVIEW

File: src/common/util/colors.js [hunk 1]
```diff
@@ -22,9 +22,15 @@ const interpolateTurbo = (value) => {
   ];
 };
 
-const getSpeedColor = (speed, maxSpeed) => {
-  const normalizedSpeed = Math.max(0, Math.min(1, speed / maxSpeed));
+const getSpeedColor = (speed, minSpeed, maxSpeed) => {
+  if (maxSpeed <= minSpeed) {
+    throw new Error('maxSpeed must be greater than minSpeed');
+  }
```
Comment (tananaev) [inline src/common/util/colors.js:28]: Why do we need this check?

File: src/map/MapRoutePoints.js [hunk 2]
```diff
@@ -57,7 +57,11 @@ const MapRoutePoints = ({ positions, onClick }) => {
   }, [onMarkerClick]);
 
   useEffect(() => {
+    if (positions.length === 0) return;
```
Comment (tananaev) [inline src/map/MapRoutePoints.js:60]: And this one also.

File: src/common/util/colors.js (hunk 1 above)
Comment (Kalabint) [reply src/common/util/colors.js]: We don't. The vars get fed correctly in the step before, and user input does not occur.

I rushed too fast once i spotted the error, this for both of them.

**Comment by Kalabint** [comment]
I just realised, that the getSpeedColor is also used in [MapRoutePath.js](https://github.com/Kalabint/traccar-web/blob/bugfix/fix-color-normalization/src/map/MapRoutePath.js), which is currently not adjusted to the new function call.

**Because of that the route line stays black in the trips menu.**
image


I already updated my [Kalabint:bugfix/fix-color-normalization](https://github.com/Kalabint/traccar-web/tree/bugfix/fix-color-normalization) branch to include the fix.

image


Should i open a new PR or what is the best way to go forward?

**Comment by tananaev** [comment]
Yes, please open a new PR.

**Pull request merged**